Added target framework support for .NET Standard 2.0

jochenwezel opened the pull request:
allows usage of library at .NET Framework, .NET Core/5/6/7/8/9, etc.

jochenwezel committed:
replaced static nunit by nuget dependency + test adapter for VS.NET

jochenwezel committed:
upgraded target frameworks to multi-target

jochenwezel committed:
cleanup old build/test pipeline

jochenwezel committed:
make .sln available in root dir for more simple execution of dotnet commands

jochenwezel committed:
added GitHub workflows

jochenwezel committed:
upgraded project files to multi-targetting

jochenwezel committed:
test for .NET Framework on windows platform

jochenwezel committed:
do not publish Jayrock .NET Framework assembly (since not yet ready for .NET Core/5/6/7/8)

jochenwezel committed:
Update Jayrock.sln

jochenwezel committed:
fixed nuget publishing

jochenwezel committed:
upgraded test project to multi-targetting

jochenwezel committed:
include README in nuget package

jochenwezel commented:
Publishing of NuGet Package still needed to change publishing target (should be easy for you to change back to your default name)

jochenwezel commented:
Unit testing still fails because NUnit2 must be upgraded to NUnit 3, first

jochenwezel commented:
Jayrock + Jayrock.Sandbox projects refer to System.Web, so they are still only targetting .NET Framework 4.8. 

.NET (Core) support can be enabled as soon as required code redesign has been completed to work around missing System.Web/classic ASP.NET

So, for now only Jayrock.Json supports .NET Standard 2.0, .NET Framework 4, .NET (Core).

jochenwezel committed:
fixed nuget package descriptions and readme

jochenwezel committed:
fix workflow since unit tests are not yet supported until upgraded to NUnit3 or higher

jochenwezel committed:
fixed nuget packaing of README

atifaziz reviewed (commented):
Thanks for this PR. Given the number of changes (129 files), this is going to take me some time to review. This is one of my low priority project. I don't know how much patience I can ask of you, but this is an interim comment to say that I'll try and get to this eventually and I'm not just ignoring it.

jochenwezel committed:
nunit3 upgade - first refactorings

jochenwezel committed:
nunit3 upgade - more refactorings considering upgrade step to nunit3 v4.x syntax

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
nunit3 v4.x upgade - more refactorings

jochenwezel committed:
fix unit test TestMethod.Issue12 for NetFramework

previously, test failed at `throw e.InnerException;` with System.Security.SecurityException : Fehler bei der Anforderung für Hauptberechtigung.

jochenwezel committed:
reverted additional tests with TimeZoneInfo

reverted partially b3a2bd752dd1707b6b16bf192d89f57df7b42cdd

jochenwezel commented:
#### NUnit test project migration

I just added the patches for migration of the test project (with some support of ChatGPT ;-))
* from NUnit2 (only support .NET Framework) to NUnit3 (supports both .NET Framework + .NET (Core). 
* Since NUnit3 v4.x introduced new syntax and dropped old syntax like e.g. Assert.AreEqual, all unit tests have got upgraded syntax for assertions

#### Found issues in unit tests
The unit tests in visual studio execute as expected, but except for
* [ ] AddsNoSerializableState (OK: .NET Framework 4.8, FAILED: .NET 8.0)
  * System.Runtime.Serialization.SerializationException : Type 'System.Collections.CollectionBase' in Assembly 'System.Collections.NonGeneric, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' is not marked as serializable.
* [ ] EmbeddedObjects (FAILED for both: .NET Framework 4.8, .NET 8.0)
  * (root)/husband/id Assert.That(reader.ReadNumber(), Is.EqualTo(expected))   Expected: 1 (Int32)   But was:  1 (JsonNumber) 
* [ ] NullPropertiesSkipped  (FAILED for both: .NET Framework 4.8, .NET 8.0)
* Assert.That(reader.ReadNumber(), Is.EqualTo(0))   Expected: 0 (Int32)   But was:  0 (JsonNumber) 
* [ ] PublicProperties  (FAILED for both: .NET Framework 4.8, .NET 8.0)
  * (root)/year Assert.That(reader.ReadNumber(), Is.EqualTo(expected))   Expected: 2000 (Int32)   But was:  2000 (JsonNumber) 
* [ ] TypeSpecific  (FAILED for both: .NET Framework 4.8, .NET 8.0)
  * (root)/owner/id Assert.That(reader.ReadNumber(), Is.EqualTo(expected))   Expected: 123 (Int32)   But was:  123 (JsonNumber) 

@atifaziz Maybe you have got some ideas what should be expected or how to solve these issues as best? (for more details, please see e.g. https://github.com/CompuMasterGmbH/CompuMaster.Jayrock/actions/runs/12494189958/job/34863545355 )

fsalas commented:
Hi, I ve used this library and is so easy to use than I love it, and being in the process of migrating to aspnet core, I gave it a try to migrate it to .net 9, modifying all to async calls in the process as core preffers, It pass all the tests except the ones for binaryserializer , I ve never done a push request or the sort, but if there is enough interest I can poolish the work and try it

atifaziz commented:
> Hi, I ve used this library and is so easy to use than I love it, and being in the process of migrating to aspnet core, I gave it a try to migrate it to .net 9, modifying all to async calls in the process as core preffers, It pass all the tests except the ones for binaryserializer , I ve never done a push request or the sort, but if there is enough interest I can poolish the work and try it

@fsalas Can you post this to the [general discussions](https://github.com/atifaziz/Jayrock/discussions/categories/general) section? Asking because it's unrelated to this PR and therefore may confuse the thread(s).

fsalas commented:
Ok, sorry for the missplaced comment
